cookbooks/aws-parallelcluster-platform/resources/fetch_dna_files.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+resource_name :fetch_dna_files
+provides :fetch_dna_files
+unified_mode true
+
+property :extra_chef_attribute_location, String, default: '/tmp/extra.json'
+
+default_action :share
+
+action :share do
+  return if on_docker?
+  return unless node['cluster']['node_type'] == 'HeadNode'
+
+  Chef::Log.info("Share extra.json with ComputeFleet")
+  ::FileUtils.cp_r(new_resource.extra_chef_attribute_location, "#{node['cluster']['shared_dir']}/dna/extra.json", remove_destination: true) if ::File.exist?(new_resource.extra_chef_attribute_location)
+
+  execute "Share dna.json with ComputeFleet" do
+    command "#{cookbook_virtualenv_path}/bin/python #{node['cluster']['scripts_dir']}/get_compute_user_data.py" \
```

Comment:
This is a step that deserves a comment to clarify. The action is about sharing dna.json with the compute fleet, but the action body is calling a script with a name that seems to only retrieve compute nodes user data.
I'm aware of the reason, but I suggest to:
1. change the action title to: Retrieve dna.json from compute user data and share with the compute fleet
2. change the script name to better reflect its goal as it does  more than simply retrieving the user data